### PR TITLE
Add support of is_compatible for old version of onnx

### PR DESCRIPTION
**Pull request by houseroad**
Fix the problem if caffe2 works with old version of onnx

### CODE REVIEW

File: caffe2/python/onnx/backend.py [hunk 1]
```diff
@@ -986,6 +986,13 @@ def supports_device(cls, device_str):
             return workspace.has_gpu_support
         return False
 
+    @classmethod
+    def is_compatible(cls, model, device='CPU', **kwargs):
+        if hasattr(super(Caffe2Backend, cls), 'is_compatible') \
+           and callable(super(Caffe2Backend, cls).is_compatible):
+            super(Caffe2Backend, cls).is_compatible(model, device, **kwargs)
```
Comment (yinghai) [inline caffe2/python/onnx/backend.py]: No return?

Comment (houseroad) [reply caffe2/python/onnx/backend.py]: Good catch, added.

File: caffe2/python/onnx/backend.py [hunk 2]
```diff
@@ -986,6 +986,14 @@ def supports_device(cls, device_str):
             return workspace.has_gpu_support
         return False
 
+    @classmethod
+    def is_compatible(cls, model, device='CPU', **kwargs):
+        if hasattr(super(Caffe2Backend, cls), 'is_compatible') \
+           and callable(super(Caffe2Backend, cls).is_compatible):
+            if not super(Caffe2Backend, cls).is_compatible(model, device, **kwargs):
```
Comment (yinghai) [inline caffe2/python/onnx/backend.py]: You can just do 
```
return super(Caffe2Backend, cls).is_compatible(model, device, **kwargs)
```

**Review by yinghai — APPROVED** [review]

File: caffe2/python/onnx/backend.py (hunk 2 above)
Comment (houseroad) [reply caffe2/python/onnx/backend.py]: Yeah, we can do that we still want to enable additional check in this function :-)

**Pull request closed without merging**